store/src/java/com/zimbra/cs/event/SolrEventStore.java

```diff
@@ -58,6 +68,71 @@ protected void deleteEventsByDataSource(String dataSourceId) throws ServiceExcep
         solrHelper.execute(accountId, req);
     }
 
+    @Override
+    public Long getContactFrequencyCount(String contact) throws ServiceException {
+        BooleanQuery.Builder searchForContactAsSenderOrReceiver = new BooleanQuery.Builder();
+        searchForContactAsSenderOrReceiver.add(getQueryToSearchContactAsSenderOrReceiver(contact), Occur.MUST);
+        if (solrHelper.needsAccountFilter()) {
+            searchForContactAsSenderOrReceiver.add(new TermQuery(new Term(LuceneFields.L_ACCOUNT_ID, accountId)), Occur.MUST);
```

Comment:
I think account ID should be a filter clause instead of a query clause, it's more efficient that way.